### PR TITLE
Wrap heartbeat detection in exception trapper

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/Amqp/ChannelDispatcher.php [hunk 1]
```diff
@@ -50,6 +50,8 @@ public function heartbeat(DeclarationManagerInterface $declarationManager)
             if (false === strpos($e->getMessage(), 'Failed to send heartbeat')) {
                 throw $e;
             }
+        } catch (ConnectionException $e) {
+            throw $e;
```
Comment (jmalloc) [inline src/Amqp/ChannelDispatcher.php]: Is something missing from here (otherwise, this is a no-op).

**Comment by jmalloc** [comment]
Am I right in thinking that this exception would then be caught in the application code, and cause a shutdown/restart without dumping a stacktrace?

**Comment by CWX-iggy** [comment]
Looks like it's working as expected. Pausing rabbitmq is sufficient for services to crash with a `HeartbeatFailureException` and restart. Fantasy's EventGroup service doesn't appear to be crashing so consistently anymore, and any crashed service recovers automatically. I've had 3 or 4 games running over the last 2 days to completion and payout. LGTM.

File: src/Amqp/ChannelDispatcher.php (hunk 1 above)
Comment (jmalloc) [reply src/Amqp/ChannelDispatcher.php]: These couple lines can still be removed, as they're simply re-throwing the same caught exception.

File: src/Amqp/ChannelDispatcher.php [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 use ErrorException;
 use Icecave\Isolator\IsolatorTrait;
+use Icecave\Overpass\Amqp\Exception\HeartbeatFailureException;
```
Comment (jmalloc) [inline src/Amqp/ChannelDispatcher.php]: This can go now too :P

**Comment by jmalloc** [comment]
I'm tagging up 2.0.3 now, just needed to do a bit of housekeeping (ditching git-flow, `develop` branch, etc).

**Pull request merged**